File: sdks/python/bolt_prediction_sdk/models/periodic_report_response.py
"""
PeriodicReportResponse 模型
"""

from typing import Optional, Any, Dict, List
from datetime import datetime
from pydantic import Field

from .base import SDKBaseModel


class PeriodicReportResponse(SDKBaseModel):
    """周期报告响应（周报/月报）"""

    report_type: str = Field(description="报告类型：weekly/monthly")
    node_id: str = Field(description="节点ID")
    node_type: str = Field(description="节点类型")
    period_start: datetime = Field(description="统计周期开始时间")
    period_end: datetime = Field(description="统计周期结束时间")
    diagnosis_summary: str = Field(description="诊断摘要")
    recommended_actions: List[str] = Field(description="推荐处置措施")
    urgency_level: str = Field(description="整体紧急程度：low/medium/high/critical")
    statistics: ReportStatisticsSchema = Field(description="统计数据")
    generated_at: datetime = Field(description="生成时间")
    model: str = Field(description="使用的模型")
    is_fallback: Optional[bool] = Field(description="是否使用降级模板", default=False)

File: sdks/java/src/main/java/com/boltprediction/sdk/model/PeriodicReportResponse.java
package com.boltprediction.sdk.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.List;
import java.util.Map;
import java.time.OffsetDateTime;

/** 周期报告响应（周报/月报） */
@JsonIgnoreProperties(ignoreUnknown = true)
public class PeriodicReportResponse {

    @JsonProperty("report_type")
    private String reportType;

    @JsonProperty("node_id")
    private String nodeId;

    @JsonProperty("node_type")
    private String nodeType;

    @JsonProperty("period_start")
    private OffsetDateTime periodStart;

    @JsonProperty("period_end")
    private OffsetDateTime periodEnd;

    @JsonProperty("diagnosis_summary")
    private String diagnosisSummary;

    @JsonProperty("recommended_actions")
    private List<String> recommendedActions;

    @JsonProperty("urgency_level")
    private String urgencyLevel;

    @JsonProperty("statistics")
    private ReportStatisticsSchema statistics;

    @JsonProperty("generated_at")
    private OffsetDateTime generatedAt;

    @JsonProperty("model")
    private String model;

    @JsonProperty("is_fallback")
    private Boolean isFallback;

    public String getReportType() {
        return reportType;
    }

    public void setReportType(String reportType) {
        this.reportType = reportType;
    }

    public String getNodeId() {
        return nodeId;
    }

    public void setNodeId(String nodeId) {
        this.nodeId = nodeId;
    }

    public String getNodeType() {
        return nodeType;
    }

    public void setNodeType(String nodeType) {
        this.nodeType = nodeType;
    }

    public OffsetDateTime getPeriodStart() {
        return periodStart;
    }

    public void setPeriodStart(OffsetDateTime periodStart) {
        this.periodStart = periodStart;
    }

    public OffsetDateTime getPeriodEnd() {
        return periodEnd;
    }

    public void setPeriodEnd(OffsetDateTime periodEnd) {
        this.periodEnd = periodEnd;
    }

    public String getDiagnosisSummary() {
        return diagnosisSummary;
    }

    public void setDiagnosisSummary(String diagnosisSummary) {
        this.diagnosisSummary = diagnosisSummary;
    }

    public List<String> getRecommendedActions() {
        return recommendedActions;
    }

    public void setRecommendedActions(List<String> recommendedActions) {
        this.recommendedActions = recommendedActions;
    }

    public String getUrgencyLevel() {
        return urgencyLevel;
    }

    public void setUrgencyLevel(String urgencyLevel) {
        this.urgencyLevel = urgencyLevel;
    }

    public ReportStatisticsSchema getStatistics() {
        return statistics;
    }

    public void setStatistics(ReportStatisticsSchema statistics) {
        this.statistics = statistics;
    }

    public OffsetDateTime getGeneratedAt() {
        return generatedAt;
    }

    public void setGeneratedAt(OffsetDateTime generatedAt) {
        this.generatedAt = generatedAt;
    }

    public String getModel() {
        return model;
    }

    public void setModel(String model) {
        this.model = model;
    }

    public Boolean getIsFallback() {
        return isFallback;
    }

    public void setIsFallback(Boolean isFallback) {
        this.isFallback = isFallback;
    }

}